axelor-account/src/main/java/com/axelor/apps/account/service/batch/BatchAutoMoveLettering.java

```diff
@@ -0,0 +1,492 @@
+
+  private static final Logger LOG = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+  protected AccountingBatchRepository accountingBatchRepository;
+  protected MoveLineRepository moveLineRepository;
+  protected CompanyRepository companyRepository;
+
+  protected MoveLineControlService moveLineControlService;
+  protected PaymentService paymentService;
+  protected ReconcileService reconcileService;
+  protected ReconcileGroupService reconcileGroupService;
+
+  protected AccountingBatch accountingBatch;
+
+  @Inject
+  public BatchAutoMoveLettering(
+      AccountingBatchRepository accountingBatchRepository,
+      MoveLineRepository moveLineRepository,
+      CompanyRepository companyRepository,
+      MoveLineControlService moveLineControlService,
+      PaymentService paymentService,
+      ReconcileService reconcileService,
+      ReconcileGroupService reconcileGroupService) {
+    super();
+    this.accountingBatchRepository = accountingBatchRepository;
+    this.moveLineRepository = moveLineRepository;
+    this.companyRepository = companyRepository;
+    this.moveLineControlService = moveLineControlService;
+    this.paymentService = paymentService;
+    this.reconcileService = reconcileService;
+    this.reconcileGroupService = reconcileGroupService;
+  }
+
+  @Override
+  protected void process() {
+    accountingBatch = batch.getAccountingBatch();
+
+    initBatch();
+
+    Map<List<Object>, Pair<List<MoveLine>, List<MoveLine>>> moveLineMap = getMoveLinesMap();
+    Comparator<MoveLine> moveLineComparator = getMoveLineComparator();
+    int reconcileMethodSelect = accountingBatch.getReconcileMethodSelect();
+
+    int i = 0;
+
+    for (Pair<List<MoveLine>, List<MoveLine>> moveLineLists : moveLineMap.values()) {
+      try {
+        findBatch();
+        accountingBatch = batch.getAccountingBatch();
+        accountingBatch.setCompany(companyRepository.find(accountingBatch.getCompany().getId()));
+        moveLineLists = moveLineService.findMoveLineLists(moveLineLists);
+        List<MoveLine> companyPartnerCreditMoveLineList =
+            moveLineLists.getLeft().stream()
+                .filter(moveLine -> moveLine.getAmountRemaining().compareTo(BigDecimal.ZERO) > 0)
+                .collect(Collectors.toList());
+        List<MoveLine> companyPartnerDebitMoveLineList =
+            moveLineLists.getRight().stream()
+                .filter(moveLine -> moveLine.getAmountRemaining().compareTo(BigDecimal.ZERO) > 0)
+                .collect(Collectors.toList());
+        companyPartnerCreditMoveLineList.sort(moveLineComparator);
+        companyPartnerDebitMoveLineList.sort(moveLineComparator);
+
+        reconcileWithMethod(
+            companyPartnerDebitMoveLineList,
+            companyPartnerCreditMoveLineList,
+            reconcileMethodSelect);
+      } catch (Exception e) {
+        TraceBackService.trace(e);
+      } finally {
+        i++;
+        if (i % AbstractBatch.FETCH_LIMIT == 0) {
```

Comment:
I think it's best to put the JPA.clear in the reconcileWithMethod, for each reconciliation done.